### DISCUSSION TITLE
How to play IP Camera RTSP feed with WebRTC in Ant Media Server?

### DISCUSSION THREAD

**Discussion by asharamseervi** [ant-media/Ant-Media-Server#4986] [FAQs] [answered]
Hey, I have an IP Camera with a built-in RTSP URL but it's not having a public IP.

I want to stream it with as low latency as possible.
Is there any way that I can play my IP camera with WebRTC using Ant Media Server?

**Comment by Mohit-3196** [comment, marked as answer]
Thank you for your question.
There are two ways to do this with Ant Media Server.

1. You can add the RTSP URL for your camera to the Ant Media Server as a stream source. You can do this as documented [here](https://antmedia.io/docs/guides/publish-live-stream/ip-cameras-and-external-resources/) or you can use [this](https://antmedia.io/rest/#/BroadcastRestService/createBroadcast) REST method by providing the source URL. With this solution, you will have video/audio data on the server side and you can record or re-stream it.

2. The second way is more innovative but it is applicable to the camera which has its own processors (mostly ARM) on it. In this solution, you can run the Embedded SDK software on your camera's processor. This software can capture video/audio data from the camera and feed that data to the WebRTC peer. In this case, data is sent directly to the WebRTC peer without passing through the server.

You can find out about Embedded SDK [here](https://antmedia.io/docs/guides/developer-sdk-and-api/sdk-integration/embedded-sdk-guide/). You can find reference project executable files for different architectures [here](https://github.com/ant-media/WebRTCEmbeddedSDKReference/tree/master/executables).

Lastly, if you want to modify and build this reference project by your self you will need SDK libraries. You can log in to [antmedia.io](https://antmedia.io/) and download the SDK libraries.

I hope it helps.
Thank you.
